x-pack/plugins/security_solution/public/resolver/view/panel.tsx

```diff
@@ -3,191 +3,245 @@
-import { useResolverDispatch } from './use_resolver_dispatch';
+import { useHistory } from 'react-router-dom';
+// eslint-disable-next-line import/no-nodejs-modules
+import querystring from 'querystring';
+import { EuiPanel } from '@elastic/eui';
+import { displayNameRecord } from './process_event_dot';
 import * as selectors from '../store/selectors';
+import { useResolverDispatch } from './use_resolver_dispatch';
+import * as event from '../../../common/endpoint/models/event';
+import { ResolverEvent } from '../../../common/endpoint/types';
+import { SideEffectContext } from './side_effect_context';
+import { ProcessEventListNarrowedByType } from './panels/panel_content_related_list';
+import { EventCountsForProcess } from './panels/panel_content_related_counts';
+import { ProcessDetails } from './panels/panel_content_process_detail';
+import { ProcessListWithCounts } from './panels/panel_content_process_list';
+import { RelatedEventDetail } from './panels/panel_content_related_detail';
+import { CrumbInfo } from './panels/panel_content_utilities';
 
-const HorizontalRule = memo(function HorizontalRule() {
-  return (
-    <EuiHorizontalRule
-      style={{
-        /**
-         * Cannot use `styled` to override this because the specificity of EuiHorizontalRule's
-         * CSS selectors is too high.
-         */
-        marginLeft: `-${euiVars.euiPanelPaddingModifiers.paddingMedium}`,
-        marginRight: `-${euiVars.euiPanelPaddingModifiers.paddingMedium}`,
-        /**
-         * The default width is 100%, but this should be greater.
-         */
-        width: 'auto',
-      }}
-    />
-  );
-});
-
-export const Panel = memo(function Event({ className }: { className?: string }) {
-  interface ProcessTableView {
-    name: string;
-    timestamp?: Date;
-    event: ResolverEvent;
-  }
+/**
+ * The team decided to use this determinant to express how we comport state in the UI with the values of the two query params:
+ *
+ * | Crumb/Table            | &crumbId                   | &crumbEvent              |
+ * | :--------------------- | :------------------------- | :----------------------  |
+ * | all processes/default  | null                       | null                     |
+ * | process detail         | entity_id of process       | null                     |
+ * | relateds count by type | entity_id of process       | 'all'                    |
+ * | relateds list 1 type   | entity_id of process       | valid related event type |
+ * | related event detail   | event_id of related event  | entity_id of process     |
+ *
+ * This component implements the strategy laid out above by determining the "right" view and doing some other housekeeping e.g. effects to keep the UI-selected node in line with what's indicated by the URL parameters.
+ *
+ * @returns {JSX.Element} The "right" table content to show based on the query params as described above
+ */
+const PanelContent = memo(function PanelContent() {
```

Comment:
Let's discuss what it would take to move this logic to the middleware.

Reply:
:reminder_ribbon: panelToShow() . `appShowedPanel` action